app/services/events/create_service.rb

```diff
@@ -92,17 +100,31 @@ def handle_persisted_event
     end
 
     def charges
-      @charges ||= event.subscription
+      event.subscription
         .plan
         .charges
         .pay_in_advance
         .joins(:billable_metric)
-        .where(invoiceable: false)
         .where(billable_metric: { code: event.code })
     end
 
-    def pay_in_advance_charges?
-      charges.any?
+    def non_invoiceable_charges
+      @non_invoiceable_charges ||= charges.where(invoiceable: false)
+    end
+
+    def invoiceable_charges
+      @invoiceable_charges ||= charges.where(invoiceable: true)
+    end
+
+    def applicable_event?
+      return true unless billable_metric.sum_agg?
+      return true unless event.properties[billable_metric.field_name]&.to_i.negative?
```

Comment:
Lint/SafeNavigationChain: Do not chain ordinary method call after safe navigation operator.